daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -462,27 +484,16 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
     }
   }
 
-  /**
-   * Creates a new instance of a variable with default value
-   */
-  def newVariableInstance(vrd: VariableRuntimeData, defaultValue: DataValuePrimitive) = {
-    val varQName = vrd.globalQName
-    val variableInstances = vTable.get(varQName)
-    Assert.invariant(variableInstances.isDefined)
-
-    variableInstances.get += vrd.createVariableInstance(VariableUtils.convert(defaultValue.getAnyRef.toString, vrd, vrd))
-  }
-
   /**
    * Creates a new instance of a variable without default value
    */
   def newVariableInstance(vrd: VariableRuntimeData) = {
     val varQName = vrd.globalQName
     val variableInstances = vTable.get(varQName)
     Assert.invariant(variableInstances.isDefined)
-
-    val v = vrd.createVariableInstance()
-    variableInstances.get += v
+    val nvi = vrd.createVariableInstance()
+    nvi.firstInstanceInitialValue = getFirstInstance(varQName).get.firstInstanceInitialValue
```

Comment:
Note that this doesn't necessarily need to come from the first instance. Since each instance gets a copy from the first, we can just copy the value from any previous instance. But could this just be
```scala
nvi.firstInstanceInitialValue = variableInstance(0).firstInstanceInitialValue
```
We don't really need ``getFirstInstance`` which is duplicating some effort, like the vtable lookup.

Reply:
You are right that we don't need getFirstInstance anymore.  I needed it before when I was setting the initial default from the NVI parser, but that has since been refactored.

That being said, I'm not sure you realized that newVariableInstance does not copy any data from the initial variable instance, so this copy is necessary.

Reply:
If we don't do the copy, then we get variables that have unset values--that feels dangerous to me. I'd rather have variables that we never use but have the right value, rather than variables that we might accientally use and have the wrong value.